task_sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -391,12 +391,55 @@ def _send_startup_message(self, ti: TaskInstance, path: str | os.PathLike[str],
         self.stdin.write(msg.model_dump_json().encode())
         self.stdin.write(b"\n")
 
-    def kill(self, signal: signal.Signals = signal.SIGINT):
+    def kill(
+        self,
+        signal_to_send: signal.Signals = signal.SIGINT,
+        escalation_delay: float = 5.0,
+        bypass_escalation: bool = False,
+    ):
+        """
+        Attempt to terminate the subprocess with a given signal.
+
+        If the process does not exit within `escalation_delay` seconds, escalate to SIGTERM and eventually SIGKILL if necessary.
+
+        :param signal_to_send: The signal to send initially (default is SIGINT).
+        :param escalation_delay: Time in seconds to wait before escalating to a stronger signal.
+        :param bypass_escalation: If True, send the signal directly to the process without escalation.
+        """
         if self._exit_code is not None:
             return
 
-        with suppress(ProcessLookupError):
-            os.kill(self.pid, signal)
+        if bypass_escalation:
```

Comment:
Lets call this `force`

Reply:
I went back and forth with it -- as i was playing around with the default.

Reply:
https://github.com/apache/airflow/pull/44465/commits/63ac85ef632c26a3a5b2f989a2c5cb804cebd3b2

Reply:
This "should" be impossible to hit, assuming it's from the wait, not the os.kill call, as even if the process has exited, the process hangs around as a Zombie (on linux this shows up as `<defunct>` in `ps aux`) until something `wait`s on it. And as long as our process is alive, nothing else should wait on it.

Reply:
it is also called here though:

https://github.com/apache/airflow/blob/840018a4264a841acbd41ca72f07938331abd3b8/task_sdk/src/airflow/sdk/execution_time/supervisor.py#L330-L339

Reply:
That's fine I think, as SIGKILL is uncatchable?

Reply:
sure, changed the approach in https://github.com/apache/airflow/pull/44465/commits/1e7dc464f5982e4b7b2b8c3a5f5585d6da53b83d